src/main/scala/components/memory-ports.scala

```diff
@@ -0,0 +1,229 @@
+// Asynchronous memory module
+
+package dinocpu
+
+import chisel3._
+import chisel3.util._
+import chisel3.internal.firrtl.Width
+
+import chisel3.util.experimental.loadMemoryFromFile
+import firrtl.annotations.MemoryLoadFileType
+
+import MemoryOperation._
+
+// A Bundle used for temporarily storing the necessary information for a write in the data memory accessor.
+class PartialWrite(val blockwidth: Int) extends Bundle {
```

Comment:
*This isn't at all necessary now*, however, at some point we'll want to extend the maskmode to set which part of the block to write.